src/scripts/controller/apiServices.ts

```diff
@@ -0,0 +1,23 @@
+import { Product, ProductResponse } from '../model/interfaces';
+
+export class ApiServices {
+    private URL: string = 'https://dummyjson.com/products';
+
+    async fetchProducts(): Promise<Product[]> {
+        const response = await fetch(`${this.URL}`);
+        const data: ProductResponse = await response.json();
+        return data.products;
+    }
+
+    async searchOneProduct(query: string): Promise<Product> {
```

Comment:
Nit.  
We can give more meaningful name here.
```suggestion
    async fetchProductById(query: string): Promise<Product> {
```

Reply:
Renamed functions name(https://github.com/krushna-simform/product-management-system/pull/3/commits/7007557a824ca530bdaa5a53df8427b0966f6a0b)